managed/ui/src/components/config/Storage/StorageConfiguration.js

```diff
@@ -149,24 +169,70 @@ class StorageConfiguration extends Component {
         break;
     }
 
-    return this.props
-      .addCustomerConfig({
-        type: 'STORAGE',
-        name: type,
-        data: dataPayload
-      })
-      .then((resp) => {
-        if (getPromiseState(this.props.addConfig).isSuccess()) {
-          // reset form after successful submission due to BACKUP_LOCATION value is shared across all tabs
-          this.props.reset();
-          this.props.fetchCustomerConfigs();
-        } else if (getPromiseState(this.props.addConfig).isError()) {
-          // show server-side validation errors under form inputs
-          throw new SubmissionError(this.props.addConfig.error);
-        }
-      });
+    if (values.type === "edit") {
+      return (
+        this.props
+        .editCustomerConfig({
+          type: 'STORAGE',
+          name: type,
+          configName: configName,
+          data: dataPayload,
+          configUUID: values.configUUID
+        })
+        .then((resp) => {
+          if (getPromiseState(this.props.editConfig).isSuccess()) {
+            this.props.reset();
+            this.props.fetchCustomerConfigs();
+          } else if (getPromiseState(this.props.editConfig).isError()) {
+            throw new SubmissionError(this.props.editConfig.error);
+          }
+        }), this.setState({
```

Comment:
not sure about this `setState()` - is this the second parameter of smth? need to fix code readability here

Reply:
I've moved it outside the return.